demoapp/demoapp/views.py

```diff
@@ -0,0 +1,22 @@
+from django.views.generic import RedirectView
+
+from demoapp.models import SubscriberEmail
+from django.shortcuts import redirect
+
+
+class NewsletterUnsubscribeView(RedirectView):
+    def get_redirect_url(self, *args, **kwargs):
+        return redirect('/')
+
+    def get(self, request, *args, **kwargs):
+        try:
+            subscriber = SubscriberEmail.objects.get(email=kwargs['email'])
+
+            if subscriber.unsubscribe_token == kwargs['token']:
+                subscriber.delete()
+                print('Successfully unsubscribed from newsletter.')
+            else:
+                print('Failed to unsubscribe: wrong token.')
+        except SubscriberEmail.DoesNotExist:
+            print('Failed to unsubscribe: unregistered address')
+        return redirect('/')
```

Comment:
No, you are inheriting RedirectView, so after validation you should let the parent class take care of the logic, ie. call the super() get() method. You would have caught the mistake in `get_redirect_url()` if you did that.